### PR TITLE
Upgrade secretjs

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: packages/polar/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "secret-polar",
-  "version": "0.9.94",
+  "version": "0.0.04",
```
Comment (udit-gulati) [inline packages/polar/package.json:3]: Fix this version

File: packages/polar/src/lib/account.ts [hunk 2]
```diff
@@ -9,51 +8,50 @@ import { getClient } from "./client";
...
-  client: CosmWasmClient;
+  client?: SecretNetworkClient;
 
-  constructor (account: Account, env: PolarRuntimeEnvironment) {
+  constructor (account: Account) {
     this.account = account;
-    this.client = getClient(env.network);
+  }
+
+  async loadClient (env: PolarRuntimeEnvironment): Promise<void> {
```
Comment (udit-gulati) [inline packages/polar/src/lib/account.ts:17]: We can name this method similar to done in junokit, ie, `setupClient()`

File: packages/polar/src/lib/account.ts [hunk 3]
```diff
@@ -9,51 +8,50 @@ import { getClient } from "./client";
...
+    this.client = await getClient(env.network);
   }
 
   async getAccountInfo (): Promise<WasmAccount | undefined> {
-    return await this.client.getAccount(this.account.address);
+    if (this.client === undefined) {
+      throw new Error("Client is not loaded, Please load client using `await loadClient(env)`");
```
Comment (udit-gulati) [inline packages/polar/src/lib/account.ts:23]: Can use a `PolarError` here.

File: packages/polar/src/lib/account.ts [hunk 4]
```diff
@@ -9,51 +8,50 @@ import { getClient } from "./client";
...
+    }
+    return await this.client.query.auth.account({ address: this.account.address });
   }
 
   async getBalance (): Promise<readonly Coin[]> {
-    const info = await this.client.getAccount(this.account.address);
-    if (info?.balance === undefined) {
+    if (this.client === undefined) {
+      throw new Error("Client is not loaded, Please load client using `await loadClient(env)`");
```
Comment (udit-gulati) [inline packages/polar/src/lib/account.ts:30]: PolarError

File: packages/polar/src/lib/deploy/contract.ts [hunk 5]
```diff
@@ -25,7 +25,7 @@ import type {
   UserAccount
 } from "../../types";
 import { loadCheckpoint, persistCheckpoint } from "../checkpoints";
-import { ExecuteResult, getClient, getSigningClient } from "../client";
+import { getClient, getSigningClient } from "../client";
```
Comment (udit-gulati) [inline packages/polar/src/lib/deploy/contract.ts:28]: Do we have an alternative to `ExecuteResult` here? Cause using `any` is not ideal.

File: packages/polar/src/lib/deploy/contract.ts [hunk 6]
```diff
@@ -259,16 +261,25 @@ export class Contract {
...
-      wasmFileContent,
-      {},
-      `upload ${this.contractName}`,
-      customFees
+    const uploadReceipt = await signingClient.tx.compute.storeCode(
+      {
+        sender: accountVal.address,
+        wasmByteCode: wasmFileContent,
+        source: "",
+        builder: ""
```
Comment (udit-gulati) [inline packages/polar/src/lib/deploy/contract.ts:269]: Can we fill some value here? Will make the upload verbose.

File: packages/polar/src/lib/deploy/contract.ts [hunk 7]
```diff
@@ -259,16 +261,25 @@ export class Contract {
...
+        gasLimit: 1000_0000_00 // TODO: Fix fees
+        // gasPriceInFeeDenom: customFees?.gas
+      }
     );
-    const codeId: number = uploadReceipt.codeId;
-    const contractCodeHash: string =
-      await signingClient.restClient.getCodeHashByCodeId(codeId);
+    const res = uploadReceipt?.arrayLog?.find((log) => log.type === "message" && log.key === "code_id");
+    if (res === undefined) {
+      throw new Error("Response for storing code not received!");
```
Comment (udit-gulati) [inline packages/polar/src/lib/deploy/contract.ts:278]: PolarError

File: packages/polar/src/lib/deploy/contract.ts [hunk 8]
```diff
@@ -338,14 +349,29 @@ export class Contract {
...
+        gasLimit: 100_000 // TODO: check gas
+      }
+    );
+
+    // Find the contract_address in the logs
+    const res = tx?.arrayLog?.find(
+      (log) => log.type === "message" && log.key === "contract_address"
+    );
+    if (res === undefined) {
+      throw new Error("Response for storing code not received!");
```
Comment (udit-gulati) [inline packages/polar/src/lib/deploy/contract.ts:372]: PolarError

**Review by udit-gulati — CHANGES_REQUESTED** [review]

File: packages/polar/src/lib/deploy/contract.ts (hunk 5 above)
Comment (coco-sha) [reply packages/polar/src/lib/deploy/contract.ts:28]: we don't

Comment (amityadav0) [reply packages/polar/src/lib/deploy/contract.ts:28]: no we don't

File: packages/polar/src/lib/deploy/contract.ts (hunk 6 above)
Comment (amityadav0) [reply packages/polar/src/lib/deploy/contract.ts:269]: can you suggest something ?

File: packages/polar/src/lib/deploy/contract.ts (hunk 5 above)
Comment (coco-sha) [reply packages/polar/src/lib/deploy/contract.ts:28]: maybe we can run this, see the value types and create a new type in type.ts
For that we need schema and cosmwasm 1.0 support.

File: packages/polar/src/lib/deploy/contract.ts (hunk 6 above)
Comment (udit-gulati) [reply packages/polar/src/lib/deploy/contract.ts:269]: We can pass `builder` and `source` as optional args to `deploy()` and pass them further here. Ref: https://secretjs.scrt.network/interfaces/MsgStoreCodeParams#source

**Review by udit-gulati — APPROVED** [review]

**Pull request merged**